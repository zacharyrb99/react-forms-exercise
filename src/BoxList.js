import {useState} from 'react';
import Box from './Box';
import NewBoxForm from './NewBoxForm';
import {v4 as uuid} from 'uuid';

const BoxList = () => {
    const INITIAL_STATE = [];

    const [boxes, setBoxes] = useState(INITIAL_STATE);

    const addBox = (newBox) => {
        setBoxes(boxes => [...boxes, {id: uuid(), ...newBox}]);
    };

    const removeBox = (id) => {
        setBoxes(boxes => boxes.filter(box => box.id !== id));
    };

    return (
        <div>
            <h2>Colored Boxes</h2>
            <NewBoxForm addBox={addBox} />
            <div className='BoxList-boxes'>
                {boxes.map(({id, width, height, color}) => <Box id={id} width={width} height={height} color={color} handleRemove={removeBox} key={id}/>)}
            </div>
        </div>
    )
}

export default BoxList;