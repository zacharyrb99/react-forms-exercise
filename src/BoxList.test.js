import {render, fireEvent} from '@testing-library/react';
import BoxList from './BoxList';

test('BoxList renders without crashing', () => {
    render(<BoxList/>);
});

test('Boxlist matches snapshot', () => {
    const {asFragment} = render(<BoxList/>);
    expect(asFragment).toMatchSnapshot();
});

test('It should add a new box', () => {
    const {queryByText, getByLabelText} = render(<BoxList/>);
    const widthInput = getByLabelText("Width:");
    const heightInput = getByLabelText("Height:");
    const colorInput = getByLabelText("Color:");
    const button = queryByText('Add Box!');

    expect(queryByText('X')).not.toBeInTheDocument();

    fireEvent.change(widthInput, {target: {value: 150} });
    fireEvent.change(heightInput, {target: {value: 150} });
    fireEvent.change(colorInput, {target: {value: 'red'} });
    fireEvent.click(button);

    expect(queryByText('X')).toBeInTheDocument();
})