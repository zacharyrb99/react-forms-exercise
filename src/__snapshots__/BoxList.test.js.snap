// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Boxlist matches snapshot 1`] = `[Function]`;
